plotly/utils.py

```diff
@@ -119,8 +119,72 @@ class PlotlyJSONEncoder(json.JSONEncoder):
 
     See PlotlyJSONEncoder.default for more implementation information.
 
+    Additionally, this encoder overrides nan functionality so that 'Inf',
+    'NaN' and '-Inf' encode to 'null'. Which is stricter JSON than the Python
+    version.
+
     """
 
+    # we want stricter JSON, so convert NaN, Inf, -Inf --> 'null'
+    nan_str = inf_str = neg_inf_str = 'null'
+
+    # uses code from official python json.encoder module. Same licence applies.
```

Comment:
this may be naive, but can't we just extend our current encoder with a check for is `NaN`, `Inf`? like how we do with datetimes https://github.com/plotly/python-api/blob/master/plotly/utils.py#L224